rfc/manifest-validation.md

```diff
@@ -0,0 +1,451 @@
+
+```ts
+export class MyChart extends Chart {
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+
+    new k8s.KubeDeployment(this, 'Deployment', {
+      spec: {
+        template: {
+          containers: [{ image: 'nginx', name: 'main' }]
+        }
+      }
+    });
+
+  }
+}
+
+const app = new App();
+new MyChart(app, 'Chart');
+app.synth();
+```
+
+Along with configuring<sup>2</sup> [datree](https://datree.io/) as the validation tool, running cdk8s synth should
+produce something similar to:
+
+```console
+❌  Ensure each container has a configured CPU request  [1 occurrence]
+    - construct: Chart/Deployment (kind: Deployment)
+💡  Missing property object `requests.cpu` - value should be within the accepted boundaries recommended by the organization
+
+❌  Ensure each container image has a pinned (tag) version  [1 occurrence]
+    - construct: Chart/Deployment (kind: Deployment)
+💡  Incorrect value for key `image` - specify an image version to avoid unpleasant "version surprises" in the future
+```
+
+> <sup>2</sup> How exactly this configuration looks like is outlined in the “How” section.
+
+### Requirements
+
+So, let’s define the requirements of this proposed validation mechanism.
+
+#### 1) Open System
+
+There are many validation tools out there. We would like cdk8s to be able to support all of them, without requiring
+code changes in the core library for every new tool. Ideally, these integrations will be maintained by the authors
+of the validation tools, and not by the cdk8s team.
+
+#### 2) Construct Aware
+
+By default, the validation tools operate on k8s resources. The reports they generate usually point to specific
+resources by name. However, when writing cdk8s applications, providing a resource name is actually discouraged,
+so the report needs to point to a specific construct instead. In addition, because of the imperative nature of
+cdk8s, it’s not always easy to figure out where exactly in the code a construct was even created.
+We would like the report to also provide a sort of traceback to help developers locate the specific code being used.
+
+#### 3) Distributable Config
+
+As mentioned earlier, we are operating under the assumption that validation config (e.g which tool, what policy)
+is defined by central compliance and security teams, not by the developers themselves. This means that these
+teams need a way to publish common configuration that all developer teams can then easily consume.
+
+## How
+
+In this section, we’ll answer the following question:
+
+1. What technical changes / additions are needed in order to fulfill all requirements?
+
+### Distributable Config
+
+Validations will be declared in the cdk8s.yaml configuration file, and the cdk8s CLI will try to
+load and execute them. For example, a configuration will look something like:
+
+```yaml
+app: node main.js
+language: typescript
+imports:
+  - k8s
+validations:
+  - http://mycompany.com/cdk8s-validations.yaml
```

Comment:
Similar concern here as for plugin policy URL, how can we validate that the source is not having malicious code within it?